### PR TITLE
`Card::Container` component - Add extra `level` args for `hover` and `active` states

### PR DESCRIPTION
**Pull request by didoo**
### :pushpin: Summary

When we implemented the `Card::Container` component (long time ago) we didn't include the specifications for the user interaction with the card (`hover`, `active` and `focused` states) because at that time we considered it only as "container" (hence, the name).

Since then the component has been used in multiple places/products for interactive cards, requiring different visual treatment for the different interaction states. Last in time, [this request](https://hashicorp.slack.com/archives/C7KTUHNUS/p1656122163303119) by @DingoEatingFuzz. As the component is built today, the only way to achieve these states would be via JS (with `onMouseOver/Enter/Leave`), which is not great.

After discussion with the team, we have acknowledged that defining all the possible "correct" states would require a proper audit, and some work on the design side to come up with the correct specs, but at the moment we don't have capacity for this. So we agreed for an intermediate solution: adding two new arguments `@levelHover` and `@levelActive` that would allow the consumers to specify via component API the elevation "level" to use on `:hover` and `:active`, delegating to their product designers which level to use for the different states.

### :hammer_and_wrench: Detailed description

In this PR I have:
- added `@levelHover` and `@levelActive` arguments to `Card::Container` component
- updated the “Component API” and the “Showcase” documentation for the `Card::Container`
  - small changes to the doc file due to linting
- updated the integration tests for `Card::Container`

Preview: https://hds-components-git-card-add-extra-level-args-hashicorp.vercel.app/components/card

#### Things to discuss 

@heatherlarsen while technically this solution works, I _think_ it would be better to expose not just the three levels `base/mid/high` for the default level, but all the possible elevation styles, `base/low/mid/high/higher/[overlay?]` and let the consumers decide which one to use (we provide guidance, not control). This would allow them to have more space for different states combinations, while at the moment using the `Card::Container` as is is very limited. What do you think?

***

### 👀 How to review

👉 Review commit-by-commit or by files changed

Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: packages/components/addon/components/hds/card/container.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ export const DEFAULT_LEVEL = 'base';
 export const DEFAULT_BACKGROUND = 'neutral-primary';
 export const DEFAULT_OVERFLOW = 'hidden';
 export const LEVELS = ['base', 'mid', 'high'];
+export const HOVER_LEVELS = ['base', 'mid', 'high'];
+export const ACTIVE_LEVELS = ['base', 'mid', 'high'];
```
Comment (alex-ju) [inline packages/components/addon/components/hds/card/container.js:9]: What is the rationale for re-defining levels for each state? Is it so we can potentially restrict the levels we allow for each state? Or do we foresee different levels for different states?

File: packages/components/app/styles/components/card/container.scss [hunk 2]
```diff
@@ -6,7 +6,10 @@
 //
 
 
+$hds-card-container-style: ( 'surface', 'elevation' );
 $hds-card-container-levels: ( 'base', 'mid', 'high' );
+$hds-card-container-hover-levels: ( 'base', 'mid', 'high' );
+$hds-card-container-active-levels: ( 'base', 'mid', 'high' );
```
Comment (alex-ju) [inline packages/components/app/styles/components/card/container.scss:12]: Same question here about redefining the levels for each state.

File: packages/components/app/styles/components/card/container.scss [hunk 3]
```diff
@@ -16,6 +19,27 @@ $hds-card-container-border-radius: 6px;
...
+      box-shadow: var(--token-#{$style}-#{$level}-box-shadow);
+    }
+  }
+  @each $level in $hds-card-container-active-levels {
+    .hds-card__container--active-level-#{$style}-#{$level}:active,
+    .hds-card__container--active-level-#{$style}-#{$level}.mock-active {
+      box-shadow: var(--token-#{$style}-#{$level}-box-shadow);
+    }
+  }
+}
```
Comment (alex-ju) [inline packages/components/app/styles/components/card/container.scss:42]: Neatly packed!

**Review by alex-ju — APPROVED** [review]
Technically all good! I'm only curious about the choice to redefine the potential levels for each state.

File: packages/components/app/styles/components/card/container.scss (hunk 2 above)
Comment (didoo) [reply packages/components/app/styles/components/card/container.scss:12]: See above.

**Comment by heatherlarsen** [comment]
@didoo I don’t recall us making a decision to limit the options to `base`, `mid`, and `high`. I'm fine leaving this open ended until we have time to do a more thorough audit.

**Comment by heatherlarsen** [comment]
@didoo Just refreshed my memory on the elevation levels...the intention for limiting these to `base`, `mid`, and `high` is:
- to align to the design choice from the visual refresh of "Intentional elevation"
  - "We use shadow with containers to elevate and group key content on the page, and to convey natural page hierarchy and modality."
- to provide a more consistent card experience
- reserve other elevation levels for specific types of components, such as `modal`, `toasts`, form controls, etc 

At this time I don't think we should be adjusting the available elevation levels, but it's a possibility that when we do a more thorough audit, a need may arise to do so. 

Screen Shot 2022-06-29 at 8 14 01 AM

File: packages/components/tests/dummy/app/templates/components/card.hbs [hunk 4]
```diff
@@ -129,4 +160,21 @@
       </div>
     </Hds::Card::Container>
   </div>
+
+  <h4 class="dummy-h4">Interaction:</h4>
```
[the remark this answers is not in the document]
Comment (didoo) [reply packages/components/tests/dummy/app/templates/components/card.hbs]: I've updated the documentation, removing these examples and improving the "How to use" section: https://hds-components-git-card-add-extra-level-args-hashicorp.vercel.app/components/card#how-to-use

File: packages/components/app/styles/components/card/container.scss (hunk 2 above)
Comment (didoo) [reply packages/components/app/styles/components/card/container.scss]: As per @heatherlarsen feedback, for now we keep the same three levels for all of them, so I have unified the arrays in a single one.

File: packages/components/addon/components/hds/card/container.js (hunk 1 above)
Comment (didoo) [reply packages/components/addon/components/hds/card/container.js]: As per @heatherlarsen feedback, for now we keep the same three levels for all of them, so I have unified the arrays in a single one.

**Review by alex-ju — APPROVED** [review]
Likewise!

**Pull request merged**